Wikipedia/UI-V5/UIViewController+WMFSearchButton.m

```diff
@@ -11,11 +11,27 @@
 #import <BlocksKit/UIBarButtonItem+BlocksKit.h>
 #import "SessionSingleton.h"
 #import "UIViewController+WMFArticlePresentation.h"
+#import "MWKSite.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 @implementation UIViewController (WMFSearchButton)
 
+static WMFSearchViewController * _sharedSearchViewController = nil;
+
++ (void)load {
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(wmfSearchButton_applicationDidEnterBackgroundWithNotification:) name:UIApplicationDidEnterBackgroundNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(wmfSearchButton_applicationDidReceiveMemoryWarningWithNotification:) name:UIApplicationDidReceiveMemoryWarningNotification object:nil];
+}
+
++ (void)wmfSearchButton_applicationDidEnterBackgroundWithNotification:(NSNotification*)note {
+    _sharedSearchViewController = nil;
```

Comment:
should we only `nil` it out if it's not in the window? e.g. if the user goes to background while on search? (same for memory warning)